### ISSUE TITLE
[a11y]: ContentSwitcher - tab navigation

### ISSUE THREAD

**Issue by VadimDez** [carbon-design-system/carbon#18362]
### Package

@carbon/react

### Browser

Chrome, Edge, Safari, Firefox

### Operating System

MacOS, Windows

### Package version

1.74.0

### Automated testing tool and ruleset

Manual testing

### Description

While working with customer, we noticed accessibility issue in `ContentSwitcher`, which allows navigation by using arrow left and right, but it is not intuitive for the visually impaired users that are using "Tab" key for the navigation across the page. Once user navigates to the ContentSwitcher -  he's also not aware how many options are there at all.
 
One possible fix that we came up is to indicate to the user in the alt text that there're multiple options and user has to navigate with arrows.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-contentswitcher--default

### Steps to reproduce

Try to navigate with Tab key

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by VadimDez** [comment]
Setting `tabIndex=0` kinda solves the problem but user has to press Tab key twice 
```tsx
<ContentSwitcher>
  <Switch tabIndex={0} name="a" text="a"/>
  <Switch tabIndex={0} name="b" text="b" />
</ContentSwitcher>
```

**Comment by finken2** [comment]
I was just looking for an issue on the secondary part mentioned here, and noticed it mentioned. With `Tabs`, voiceover tells me "blah selected, tab, group, 3 of 10" whereas with ContentSwitcher it's always "1 of 1".  Hopefully that's an easy-ish fix and just got lost in this issue's bigger tab navigation problem. 

In storybook, you'd expect to hear "2 of 3", but always "1 of 1":

<img width="695" height="375" alt="Image" src="https://github.com/user-attachments/assets/5712559a-39c6-43f9-bf2d-d3ce62a420ff" />

@tay1orjones Let me know if you want me to split this off into a separate issue.
